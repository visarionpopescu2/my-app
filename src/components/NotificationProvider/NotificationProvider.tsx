import { ReactNode } from 'react';
import NotificationContext from 'context/NotificationContext/NotificationContext';
import { NOTIFICATION_TYPE } from 'types';
import { notification } from 'antd';

function NotificationProvider(props: { children: ReactNode }) {
  const [api, contextHolder] = notification.useNotification();
  
  const openNotification = (message: string, type: NOTIFICATION_TYPE) => {
    api[type]({ message, placement: 'bottomRight' });
  }
  
  return (
    <NotificationContext.Provider value={{ openNotification }}>
      {props.children}
      {contextHolder}
    </NotificationContext.Provider>
  )
}

export default NotificationProvider;
