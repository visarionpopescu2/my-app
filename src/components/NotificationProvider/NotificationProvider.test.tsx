import { render } from '@testing-library/react';
import NotificationProvider from './NotificationProvider';

describe('NotificationProvider', () => {
  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should render NotificationProvider component', () => {
    let view = render(
      <NotificationProvider>
        <div>children</div>
      </NotificationProvider>
    )
    expect(view.asFragment()).toMatchSnapshot();
  });
});
