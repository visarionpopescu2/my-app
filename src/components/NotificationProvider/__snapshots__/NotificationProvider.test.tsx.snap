// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationProvider should render NotificationProvider component 1`] = `
<DocumentFragment>
  <div>
    children
  </div>
</DocumentFragment>
`;
